pyomo/contrib/solver/base.py

```diff
@@ -393,8 +406,26 @@ def _map_config(
             self.config.time_limit = timelimit
         if report_timing is not NOTSET:
             self.config.report_timing = report_timing
-        if options is not NOTSET:
+        if hasattr(self, 'options'):
+            self.config.solver_options.set_value(self.options)
```

Comment:
Looking for a "possibly defined" attribute seems weird.  I think I would prefer always defining a reasonable attribute (either None, or an empty dict or an empty`Bunch`), and if the container is not empty then pass it to the ConfigDict.